cachito/workers/pkg_managers/gomod.py

```diff
@@ -446,7 +472,7 @@ def go_list_deps(pattern: Literal["./...", "all"]) -> Iterator[GoPackage]:
                 elif not dep.module or dep.module.main:
                     # Standard=false, Module.Main=true
                     # Standard=false, Module=null   <- probably a to-be-generated package
-                    dep_version = main_module_version
+                    dep_version = local_modules.main.version
```

Comment:
The version will probably be wrong for packages coming from workspaces other than the one selected by the user. It seems to be handled correctly in cachi2 though, so I wouldn't mind pretending I never noticed this :smile:

Reply:
Either that or maybe we don't report such packages at all, since as far as I can tell the `./...` pattern isn't really workspace-aware https://github.com/containerbuildsystem/cachi2/pull/553#discussion_r1654667805. Anyway, to-be-fixed in cachi2?

Reply:
So I did a few tests targeting our [test repo](https://github.com/cachito-testing/cachi2-gomod/tree/go_workspaces), and here's what I found:

The workspace packages are only identified by `go list ./...` when the command is run inside the worskpace folder. That means that Cachito will only report them if the workspace was specified as an input package (in which case, the version will end up being correct, since it correspond to the actual parent module).

In case the workspace containing packages was not specified as an input package, the package won't be reported at all.

A quick test to check this:
```
$ curl -H "content-type: application/json" -d '{
  "repo": "https://github.com/cachito-testing/cachi2-gomod",
  "ref": "b97aca303b29dca71b89185da6935249b0deef68",
  "pkg_managers": ["gomod"],
  "packages": {
    "gomod": [{"path": "workspace_modules/echo"}, {"path": "workspace_modules/hi/hiii"}, {"path": "workspace_modules/hello"}]
  }
}' localhost:8080/api/v1/requests

$ curl -s localhost:8080/api/v1/requests/{request_id} | grep "echo/v4/middleware"
   "name": "github.com/labstack/echo/v4/middleware",
```
When we remove the echo workspace from the list of input packages, the `echo/v4/middleware` package is absent:
```
$ curl -H "content-type: application/json" -d '{
  "repo": "https://github.com/cachito-testing/cachi2-gomod",
  "ref": "b97aca303b29dca71b89185da6935249b0deef68",
  "pkg_managers": ["gomod"],
  "packages": {
    "gomod": [ {"path": "workspace_modules/hi/hiii"}, {"path": "workspace_modules/hello"}]
  }
}' localhost:8080/api/v1/requests

$ curl -s localhost:8080/api/v1/requests/{request_id} | grep "echo/v4/middleware"
  
```

Reply:
I agree we should fix it in Cachi2, then backport the solution here.